pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go

```diff
@@ -133,6 +133,13 @@ func (r *Reconciler) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 			Name: "Deploying Shoot namespace in Seed",
 			Fn:   flow.TaskFn(botanist.DeploySeedNamespace).RetryUntilTimeout(defaultInterval, defaultTimeout),
 		})
+		// TODO(vicwicker): Remove after Gardener v1.104 got released.
+		_ = g.Add(flow.Task{
+			Name:         "Reconciling kube-state-metrics for Shoot in Seed for the monitoring stack during migration to Gardener v1.104",
+			Fn:           flow.TaskFn(botanist.DeployKubeStateMetrics).RetryUntilTimeout(defaultInterval, 2*time.Minute),
```

Comment:
```suggestion
			Fn:           flow.TaskFn(func(ctx context.Context) error {
				// check here if the `kube-state-metrics` deployment exists in the shoot namespace. if not, return nil
				return botanist.DeployKubeStateMetrics(ctx)
			}).RetryUntilTimeout(defaultInterval, 2*time.Minute),
```

We don't want to prematurely deploy KSM into the shoot namespace if it does not exist yet. We just want to migrate it early if it already exists. Please adapt :)

Reply:
It should be addressed now.